### ISSUE TITLE
Database Implementation Issue

### ISSUE THREAD

**Issue by utkarshjain12042004** [utkarshjain12042004/IREG-Image-Registartion-Based-Attendance-Mangement-System#7]
While working on the start attendance page, I decided to create 5 separate rows each having 6 columns. One column designated for the day of the week, and the others designated for the 5 periods in the day. I did not create separate pages to add in the teacher details, suject details etc. therefore, I had to manually add it to the database. While adding the data, I was constantly receiving an error while adding data in the tbl_class_timetable(this table will have the timetable for the students and the teachers) and the tbl_student_attendance(this will store the student attendance for a class). While adding data in these 2 tables, I was getting an error stating Error Code: 1452. Cannot add or update a child row: a foreign key constraint fails. This is preventing me from adding data to these 2 tables

**Comment by utkarshjain12042004** [comment]
Bug Description: I have finally resolved the error. The error was arising due to the foreign keys of both the tables. I had created 2 foreign keys relating to these tables. Timetable_Period_Number connecting the tbl_student_attendance to tbl_class_timetable; and Class_ID connecting the tbl_class_timetable to tbl_student_attendance. These foreign keys were creating a sort of circle. Timetable_Period_Number field in tbl_student_attendance was the parent field and the Period_Number field in the tbl_class_timetable was the child field. Also, the Class_ID field in tbl_class_timetable  was the parent field and the Timetable_Class_ID field in tbl_student_attendance was the child field. Essentially, I need to fill up the Timetable_Period_Number field in tbl_student_attendance first before adding data in the Period_Number field in the tbl_class_timetable . Also, I had to fill up the Class_ID field in tbl_class_timetable before filling up the Timetable_Class_ID field in tbl_student_attendance. This would become impossible because adding data in both the table at the same time is impossible considering all the columns were set to not null.
